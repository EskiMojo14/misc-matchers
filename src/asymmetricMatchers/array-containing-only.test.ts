describe("arrayContainingOnly", () => {
  it("should expect an array to contain only the expected values", () => {
    expect([1, 2, 2, 3]).toEqual(expect.arrayContainingOnly([1, 2, 3]));

    expect(() => {
      expect([1, 2, 2, 3]).toEqual(expect.arrayContainingOnly([1, 2]));
    }).toThrowErrorMatchingSnapshot();

    expect([1, 2, 2, 3]).toEqual(expect.not.arrayContainingOnly([1, 2]));

    expect(() => {
      expect([1, 2, 2, 3]).toEqual(expect.not.arrayContainingOnly([1, 2, 3]));
    }).toThrowErrorMatchingSnapshot();

    // vs
    expect([1, 2, 2, 3]).toEqual(expect.arrayContaining([1, 2]));
  });
  it("should throw an error if the expected value is not an array", () => {
    expect(() => {
      // @ts-expect-error testing invalid input
      expect([]).toEqual(expect.arrayContainingOnly(1));
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw an error if the received value is not an array", () => {
    expect(() => {
      expect(1).toEqual(expect.arrayContainingOnly([]));
    }).toThrowErrorMatchingSnapshot();
  });
});
