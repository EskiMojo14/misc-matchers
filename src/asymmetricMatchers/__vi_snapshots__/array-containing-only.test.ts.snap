// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`arrayContainingOnly > should expect an array to contain only the expected values 1`] = `expected [ 1, 2, 2, 3 ] to deeply equal arrayContainingOnly<1,2>`;

exports[`arrayContainingOnly > should expect an array to contain only the expected values 2`] = `expected [ 1, 2, 2, 3 ] to deeply equal not.arrayContainingOnly<1,2,3>`;

exports[`arrayContainingOnly > should throw an error if the expected value is not an array 1`] = `
<d>expect.</b>arrayContainingOnly<d>(</b><g>1</g><d>)</b>

<b>Matcher error</b>: <g>Expected</g> value must be an array

Expected has type:  number
Expected has value: 1
`;

exports[`arrayContainingOnly > should throw an error if the received value is not an array 1`] = `expected 1 to deeply equal arrayContainingOnly<>`;
